django/contrib/admin/filters.py

```diff
@@ -264,26 +264,47 @@ def choices(self, changelist):
 class ChoicesFieldListFilter(FieldListFilter):
     def __init__(self, field, request, params, model, model_admin, field_path):
         self.lookup_kwarg = '%s__exact' % field_path
+        self.lookup_kwarg2 = '%s__isnull' % field_path
         self.lookup_val = request.GET.get(self.lookup_kwarg)
+        self.lookup_val2 = request.GET.get(self.lookup_kwarg2)
         super(ChoicesFieldListFilter, self).__init__(
             field, request, params, model, model_admin, field_path)
 
     def expected_parameters(self):
-        return [self.lookup_kwarg]
+        return [self.lookup_kwarg, self.lookup_kwarg2]
 
     def choices(self, changelist):
         yield {
             'selected': self.lookup_val is None,
-            'query_string': changelist.get_query_string({}, [self.lookup_kwarg]),
+            'query_string': changelist.get_query_string(
+                {}, [self.lookup_kwarg, self.lookup_kwarg2]
+            ),
             'display': _('All')
         }
+        has_none = False
+        none_title = ''
```

Comment:
I guess we could eliminate the `has_none` variable and simply check `if none_title` -- I don't think a user would define an expected title of an empty string.